helm/install/templates/manager.yaml

```
@@ -29,4 +29,6 @@ spec:
         - name: PGO_TARGET_NAMESPACE
           valueFrom: { fieldRef: { apiVersion: v1, fieldPath: metadata.namespace } }
         {{- end }}
+        securityContext:
+          {{- toYaml .Values.containerSecurityContext | nindent 10 }}
```

This shouldn't be configurable. I would just set the parameters that we want in it:

```suggestion
            allowPrivilegeEscalation: false
            readOnlyRootFilesystem: true
            runAsNonRoot: true
```

---

To be honest I understand that it would be the most secure option but do not understand why you wrote that. You can even customize it with kustomize and we have no chance to stop if the user wants to override that. I would give that freedom option here as well.

---

That is a difference between Kustomize and Helm where one can choose to override it, but in an ideal world, this should not be overridden. The Operator does not need elevated container privileges.

I'd be happy to commit this with the requested changes.

---

Agreed, updated the PR  1655388

---

:+1: